src/senaite/databox/browser/view.py

```diff
@@ -57,6 +57,7 @@
 class DataBoxView(ListingView):
     """The default DataBox view
     """
+    from senaite.core.api import dtime
```

Comment:
Does it have a reason why you added the import at class level?

Reply:
I recall the reason was that `dtime` was unreachable from a template expression statement, if import statement declared for module level..

Would you think its better to return import statement to the module level and provide sort of wrapper from `DataBoxView` class to that template expression? Or even somehow import `dtime` right in template (is that possible?)?

Reply:
Ok, that's strange... Maybe this is because of restricted Python, but not sure.
Anyhow, I think that accessing the module directly is bad style and should be avoided.
Better create two more property methods in the class, that return the right value for you, e.g.:

```python
@property
def date_from(self):
    if not self.context.date_from:
        return ""
    return dtime.date_to_string(self.context.date_from)
```

And use then this in your template:

```html
<input type="date"
       class="form-control"
       tal:attributes="value python:view.date_from"
       name="senaite.databox.date_from"/>
```

Reply:
Thanks! I've added properties.

Also, I've added `date_to > date_from` comparison, which should make report behaviour more intuitive. In case user puts `date_to` earlier than date_from - we use `[date_from, date_from]` range for date query.